packages/story-editor/src/app/media/uploadQueue/index.ts

```diff
@@ -110,26 +120,33 @@ function useMediaUploadQueue() {
 
   // Try to update placeholder resources for freshly transcoded file if still missing.
   useEffect(() => {
-    (async () => {
-      await Promise.all(
-        state.queue.map(async (item) => {
+    void (async () => {
+      void (await Promise.all(
+        state.queue.map(async (item: QueueItem) => {
           const { id, file, state: itemState, resource } = item;
           if (
             ![
-              ITEM_STATUS.TRIMMED,
-              ITEM_STATUS.MUTED,
-              ITEM_STATUS.CROPPED,
-              ITEM_STATUS.TRANSCODED,
+              ItemStatus.Trimmed,
+              ItemStatus.Muted,
+              ItemStatus.Cropped,
+              ItemStatus.Transcoded,
             ].includes(itemState) ||
             !resource.isPlaceholder ||
-            resource.poster
+            (resource.type === ResourceType.Video && resource.poster)
           ) {
             return;
           }
 
           try {
             const { resource: newResource, posterFile } =
-              await getResourceFromLocalFile(file);
+              (await getResourceFromLocalFile(file)) as {
+                resource:
+                  | ImageResource
+                  | VideoResource
+                  | GifResource
+                  | AudioResource;
+                posterFile: File | null;
```

Comment:
Why is this assertion here needed? Isn't the return type of `getResourceFromLocalFile` already implied?